### ISSUE TITLE
Validation of CosemDataset attributes raises errors

### ISSUE THREAD

**Issue by federico-carrara** [tlambert03/microsim#66]
### Description

While instantiating the `Simulation` in `examples/notebooks/cosem.ipynb` I encountered the following pydantic Validation Error:
```
ValidationError: 2 validation errors for CosemLabel
sample.organism
  Input should be a valid list [type=list_type, input_value=None, input_type=NoneType]
    For further information visit https://errors.pydantic.dev/2.7/v/list_type
image_acquisition.start_date
  Input should be a valid datetime [type=datetime_type, input_value=None, input_type=NoneType]
    For further information visit https://errors.pydantic.dev/2.7/v/datetime_type
```

### What I Did

I went on to do some debugging, and I realized that the validation error most likely comes from this [line](https://github.com/tlambert03/microsim/blob/main/src/microsim/cosem/_client.py#L185).

Specifically, for some entries (i.e., dictionaries) in the `response.data` data structure, the values for the keys `sample.organism` and `image_acquisition.start_date` are `None`. This creates a mismatch with the declared types for `CosemDataset` attributes. 

### Temporary Fix

[Here](https://github.com/tlambert03/microsim/blob/fb676d2f255461ae34a44129667036dc64c4928f/src/microsim/cosem/models.py#L56) and [here](https://github.com/tlambert03/microsim/blob/fb676d2f255461ae34a44129667036dc64c4928f/src/microsim/cosem/models.py#L63) we can also accept `None` as an alternative type.

I would be happy to discuss better ways to fix this issue :)

**Comment by tlambert03** [comment]
Thanks Federico!Yes let’s update the models to make those two fields optional, to reflect the data that we’re now getting from the server.On Jul 11, 2024, at 5:06 AM, Federico Carrara ***@***.***> wrote:﻿
Description
While instantiating the Simulation in examples/notebooks/cosem.ipynb I encountered the following pydantic Validation Error:
ValidationError: 2 validation errors for CosemLabel
sample.organism
  Input should be a valid list [type=list_type, input_value=None, input_type=NoneType]
    For further information visit https://errors.pydantic.dev/2.7/v/list_type
image_acquisition.start_date
  Input should be a valid datetime [type=datetime_type, input_value=None, input_type=NoneType]
    For further information visit https://errors.pydantic.dev/2.7/v/datetime_type

What I Did
I went on to do some debugging, and I realized that the validation error most likely comes from this line.
Specifically, for some entries (i.e., dictionaries) in the response.data data structure, the values for the keys sample.organism and image_acquisition.start_date are None. This creates a mismatch with the declared types for CosemDataset attributes.
Temporary Fix
Here and here we can also accept None as an alternative type.
I would be happy to discuss better ways to fix this issue :)
